### ISSUE TITLE
Bug in parsing of nested integer selectors

### ISSUE THREAD

**Issue by math1985** [mapbox/carto#369]
Consider a project.mml consisting of a single layer with the following SQL query:
```(select way, name, 1 as int from planet_osm_polygon) as data```

Now compare the following two code samples:

```
[int < 22222] {
  text-name: "[name]";
  text-face-name: "DejaVu Sans Book";
  text-size: 12;
  [int > 2223] { text-size: 13; }
}
```

```
[int < 22222] {
  text-name: "[name]";
  text-face-name: "DejaVu Sans Book";
  text-size: 12;
  [int > 2222] { text-size: 13; }
}
```

The examples only differ in the second value, which is 2222 versus 2223.

The first sample gives an error: ```style.mss:5:24 Property text-name required for defining text styles.``` The second sample parses fines.

More in general, it seems that an error is generated whenever the second value has one digit less than the first value, and the second value is larger than the first value of which the last digit is truncated.

**Comment by math1985** [comment]
This bug is still blocking several important improvements in openstreetmap-carto, such as https://github.com/gravitystorm/openstreetmap-carto/issues/1028. It would be great if somebody could try to fix this issue!

**Comment by nebulon42** [comment]
@math1985 I tried to reproduce it (current master), but was not able to. I created a test.mss file and fed it directly into carto. I guess whether there is a MML file should make no difference. Output was the following:

````
<Style name="style" filter-mode="first">
  <Rule>
    <Filter>([int] &lt; 22222) and ([int] &gt; 2223)</Filter>
    <TextSymbolizer size="13" face-name="DejaVu Sans Book" ><![CDATA[[name]]]></TextSymbolizer>
  </Rule>
  <Rule>
    <Filter>([int] &lt; 22222)</Filter>
    <TextSymbolizer face-name="DejaVu Sans Book" size="12" ><![CDATA[[name]]]></TextSymbolizer>
  </Rule>
</Style>
````

Could you please verify if it still fails for you?

**Comment by math1985** [comment]
Confirmed that this is fixed by ad39e0ba227e0bfb179d1dfee304a6e8e18fb210.
